satpy/tests/reader_tests/test_ahi_gridded.py

```diff
@@ -0,0 +1,262 @@
+
+        # Check that exception is raised if bad calibration is passed
+        with self.assertRaises(NotImplementedError):
+            self.fh.calibrate(in_data, 'lasers')
+
+        # Check that exception is raised if no file is present
+        np_loadtxt.side_effect = FileNotFoundError
+        with self.assertRaises(FileNotFoundError):
+            self.fh.calibrate(in_data, 'reflectance')
+
+
+class TestAHIGriddedFileHandler(unittest.TestCase):
+    """Test case for the file reading."""
+    def new_unzip(fname):
+        """Fake unzipping."""
+        if fname[-3:] == 'bz2':
+            return fname[:-4]
+        else:
+            return fname
+
+    @mock.patch('satpy.readers.ahi_gridded.unzip_file',
+                mock.MagicMock(side_effect=new_unzip))
+    def setUp(self):
+        """Create a test file handler."""
+        m = mock.mock_open()
+        with mock.patch('satpy.readers.ahi_gridded.open', m, create=True):
+            in_fname = 'test_file.bz2'
+            fh = AHIGriddedFileHandler(in_fname,
+                                       {'area': 'fld'},
+                                       filetype_info={'file_type': 'tir.01'})
+
+            # Check that the filename is altered for bz2 format files
+            self.assertNotEqual(in_fname, fh.filename)
+            self.fh = fh
+
+        key = {'calibration': 'counts',
+               'name': 'vis.01'}
+        info = {'units': 'unitless',
+                'standard_name': 'vis.01',
+                'wavelength': 10.8,
+                'resolution': 0.05}
+        self.key = key
+        self.info = info
+
+    @mock.patch('satpy.readers.ahi_gridded.np.memmap')
+    def test_dataread(self, memmap):
+        """Check that a dask array is returned from the read function."""
+        test_arr = np.zeros((10, 10))
+        memmap.return_value = test_arr
+        m = mock.mock_open()
+        with mock.patch('satpy.readers.ahi_gridded.open', m, create=True):
+            res = self.fh._read_data(mock.MagicMock())
+            np.testing.assert_allclose(res, da.from_array(test_arr))
+
+    @mock.patch('satpy.readers.ahi_gridded.AHIGriddedFileHandler._read_data')
+    def test_get_dataset(self, mocked_read):
+        """Check that a good dataset is returned on request."""
+        m = mock.mock_open()
+
+        out_data = np.array([[100., 300., 500.],
+                             [800., 1500., 2040.]])
+        mocked_read.return_value = out_data
+        with mock.patch('satpy.readers.ahi_gridded.open', m, create=True):
+            res = self.fh.get_dataset(self.key, self.info)
+            mocked_read.assert_called()
+            # Check output data is correct
+            np.testing.assert_allclose(res.values, out_data)
+            # Also check a couple of attributes
+            self.assertEqual(res.attrs['name'], self.key['name'])
+            self.assertEqual(res.attrs['wavelength'], self.info['wavelength'])
+
+
+class TestAHIGriddedLUTs(unittest.TestCase):
```

Comment:
Ok, I think this testcase need to become a bit more concrete that just checking if some functions are called.

I would propose mocking the ftplib so as to create a dummy tar file (on the fly) in a given directory and then check that then other functions really do untar, remove, etc.